pnet_datalink/src/pcap.rs

```diff
@@ -24,19 +24,25 @@ pub struct Config {
     pub read_timeout: Option<Duration>,
 }
 
-impl<'a> From<&'a datalink::Config> for Config {
-    fn from(config: &datalink::Config) -> Config {
-        Config{
+impl<'a> From<&'a super::Config> for Config {
+    fn from(config: &super::Config) -> Config {
+        let mut c = Config{
             read_buffer_size: config.read_buffer_size,
             read_timeout: config.read_timeout,
+        };
+        // pcap buffer size must be at least 65616; if someone set
+        // the buffer too small, just override it with the default.
+        if c.read_buffer_size <= 65616 {
```

Comment:
I'd rather not do this - if you specify a buffer too small, you should get an error in `datalink::channel()` instead, and the developer should explicitly set a better buffer size.

Reply:
Not a bad idea. It would also be nice to find a way to plumb through the backend-specific defaults with the abstract global defaults.